### ISSUE TITLE
 Model evaluation error

### ISSUE THREAD

**Issue by binglinchengxiash** [tech-srl/code2vec#125]
When the model is evaluated, an error is reported using java-small dataset.
![image](https://user-images.githubusercontent.com/15050040/125254712-4d02e180-e32d-11eb-9019-b7e2a149b0f8.png)

**Comment by urialon** [comment]
Hi @binglinchengxiash , thank you for your interest in our work!

It seems that the denominator is zero. This can be fixed by adding a line such as:
```
if self.nr_true_positives + self.nr_false_positives == 0:
   return 0
```

But I wonder why did it happen,  it might hint that the test set is empty? Do you know whether there are any examples in the test/validation set?

**Comment by binglinchengxiash** [comment]
> Hi @binglinchengxiash , thank you for your interest in our work!
> 
> It seems that the denominator is zero. This can be fixed by adding a line such as:
> 
> ```
> if self.nr_true_positives + self.nr_false_positives == 0:
>    return 0
> ```
> 
> But I wonder why did it happen, it might hint that the test set is empty? Do you know whether there are any examples in the test/validation set?

I use astminer to generate data for feeding code2vec.And the dataset is java-small. I found all results are zeros.I am not sure the reason.
![image](https://user-images.githubusercontent.com/15050040/125570802-1869aeed-51ba-47a2-b310-95aa25064cdc.png)

**Comment by binglinchengxiash** [comment]
> Hi @binglinchengxiash , thank you for your interest in our work!
> 
> It seems that the denominator is zero. This can be fixed by adding a line such as:
> 
> ```
> if self.nr_true_positives + self.nr_false_positives == 0:
>    return 0
> ```
> 
> But I wonder why did it happen, it might hint that the test set is empty? Do you know whether there are any examples in the test/validation set?

I use astminer to generate data for feeding code2vec.And the dataset is java-small. I found all results are zeros.I am not sure the reason.
![image](https://user-images.githubusercontent.com/15050040/125571452-3e5d6db4-9f7c-483d-a74c-a2208688bc87.png)

**Comment by urialon** [comment]
Hi @binglinchengxiash ,
Sorry for the delayed response.

Have you managed to solve this?
This may be a problem with astminer.

Best,
Uri

**Comment by binglinchengxiash** [comment]
Thank you, I have solved my problems.

**Comment by WQR53** [comment]
Hi @binglinchengxiash ,

I met the same problem that PRECISION, RECALL, and F1 were always zero when training.
"
2022-05-03 01:26:07,479 INFO     After 12 epochs -- top10_acc: [0.32272727 0.51818182 0.61363636 0.66818182 0.69090909 0.73636364 0.75454545 0.76363636 0.76818182 0.79545455], precision: 0.0, recall: 0.0, F1: 0
2022-05-03 01:26:16,707 INFO     Average loss at batch 100: 0.008121,   throughput: 399 samples/sec
"
Do you know what causes this problem？What should I do to fix it?

I would be very grateful indeed for any help you could give me.

**Comment by WQR53** [comment]
Hi @urialon ,
I opened a new issue [here ](https://github.com/tech-srl/code2vec/issues/151). Thank you!
